core/src/main/scala/cats/package.scala

```diff
@@ -68,6 +68,16 @@ package object cats {
     def apply[A](a: A): Id[A] = a
   }
 
+  /**
+   * Like `Nothing` but works better with type inference.
+   */
+  type INothing <: Nothing
+
+  /**
+   * Like `INothing` but as a type constructor.
+   */
+  type Pure[A] <: Nothing
```

Comment:
can we add some doctest example here? I think we can assume people know what Nothing is useful for, but I don't know of uses for `Pure[A]` looking at this.

Reply:
I've added a doctest example - it's fairly trivial, but hopefully goes some way to address @SystemFw's concern that `Pure` is in principle specific to fs2.

Reply:
I'm not sure it does, since this compiles:
```scala
  type Alg[F[_], A] = Either[F[A], A]
  val pureAlg: Alg[Nothing, Int] = Right(3)
  val optAlg: Alg[Option, Int] = pureAlg
  val listAlg: Alg[List, Int] = pureAlg
```
so why is it named `Pure`? why is it an inequality with `Nothing`? If all you need is covariance (also address @armanbilge point about covariance being the important aspect here), `Nothing` is enough, the specific shape we came to with `Pure` is dictated by other constraints in fs2's api specifically

Reply:
Yeah, this occurred to me afterwards. I'm trying to come up with a reasonably generic example that won't compile with `Nothing` but so far without success - this inclines me towards agreeing with you, but also makes me more sad that we have to expose `fs2.Pure` rather than `Nothing` in the http4s api.

Reply:
Yes, thanks, this answered a lot of my questions/confusion. Now I see what you are getting at when you say
> the specific shape we came to with `Pure` is dictated by other constraints in fs2's api specifically